test/functional/fixtures/before-test.js

```diff
@@ -0,0 +1,128 @@
+var browserNatives     = require('testCafe-browser-natives');
+var SauceLabsConnector = require('saucelabs-connector');
+var Promise            = require('pinkie');
+var caller             = require('caller');
+var path               = require('path');
+var createTestCafe     = require('../../../lib');
+var config             = require('../config.js');
+var Site               = require('../site');
+var getTestError       = require('../helpers/report-helper.js');
+
+var testCafe     = null;
+var browsersInfo = null;
+
+var slConnector = null;
+var slBrowsers  = null;
+
+before(function () {
+    return createTestCafe(config.testCafe.hostname, config.testCafe.port1, config.testCafe.port2)
+        .then(function (tc) {
+            var openBrowserPromises = null;
+
+            testCafe = tc;
+
+            browsersInfo = config.browsers

```

Comment:
I have the feeling that most of this code can be refactored into named functions